src/scanner.cc

```diff
@@ -48,6 +49,17 @@ struct Scanner {
     lexer->advance(lexer, false);
   }
 
+  bool advance_matching_and_whitespace(TSLexer *lexer, const char *match) {
+    for( unsigned i = 0; i < strlen(match); i += 1 ){
```

Comment:
I think the `strlen` call is redundant because you're already iterating through the null-terminated string character by character. You can just use the null terminator as your termination condition. How about this:

```c
for (const char *c = match; *c; c++) {
  if (lexer->lookahead == *c) {
    // ...
  }
}
```